publish/aws_parameter_store.go

```diff
@@ -0,0 +1,136 @@
+		if strings.HasSuffix(awspsd.parameterPath, "/") {
+			return awspsd.parameterPath + fileName
+		}
+		return awspsd.parameterPath
+	}
+	// Default: use filename as parameter name
+	if !strings.HasPrefix(fileName, "/") {
+		return "/" + fileName
+	}
+	return fileName
+}
+
+// Upload returns NotImplementedError as AWS Parameter Store does not support uploading encrypted files directly.
+func (awspsd *AWSParameterStoreDestination) Upload(fileContents []byte, fileName string) error {
+	return &NotImplementedError{"AWS Parameter Store does not support uploading encrypted sops files directly. Use UploadUnencrypted instead."}
+}
+
+// UploadUnencrypted uploads unencrypted data to AWS Parameter Store as JSON.
+func (awspsd *AWSParameterStoreDestination) UploadUnencrypted(data map[string]interface{}, fileName string) error {
+	ctx := context.TODO()
+
+	// Load AWS config - use explicit region if provided, otherwise rely on SDK defaults
+	var cfg aws.Config
+	var err error
+	if awspsd.region != "" {
+		cfg, err = config.LoadDefaultConfig(ctx, config.WithRegion(awspsd.region))
+	} else {
+		cfg, err = config.LoadDefaultConfig(ctx)
+	}
+	if err != nil {
+		return fmt.Errorf("unable to load AWS SDK config: %w", err)
+	}
+
+	client := ssm.NewFromConfig(cfg)
+	parameterName := awspsd.Path(fileName)
+
+	// Convert data to JSON string for storage
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("failed to marshal data to JSON: %w", err)
+	}
+	parameterValue := string(jsonData)
+
+	// Check if parameter already exists and compare content
+	getParamOutput, err := client.GetParameter(ctx, &ssm.GetParameterInput{
+		Name:           aws.String(parameterName),
+		WithDecryption: aws.Bool(true), // Decrypt for comparison if it's a SecureString
+	})
+
+	parameterExists := true
+	if err != nil {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) && apiErr.ErrorCode() == "ParameterNotFound" {
+			parameterExists = false
+			parameterLog.Infof("Parameter %s does not exist, will create new parameter", parameterName)
+		} else {
+			parameterLog.Warnf("Cannot check if destination parameter already exists in %s. New version will be created even if the data has not been changed.", parameterName)
+		}
+	}
+
+	// If parameter exists, check if content is identical
+	if parameterExists && getParamOutput.Parameter.Value != nil {
+		if *getParamOutput.Parameter.Value == parameterValue {
+			parameterLog.Infof("Parameter %s is already up-to-date.", parameterName)
+			return nil
+		}
+	}
+
+	// Always use SecureString for security - SOPS files may contain secrets
+	paramType := types.ParameterTypeSecureString
+
+	// Put parameter (creates or updates)
+	_, err = client.PutParameter(ctx, &ssm.PutParameterInput{
+		Name:        aws.String(parameterName),
+		Value:       aws.String(parameterValue),
+		Type:        paramType,
+		Overwrite:   aws.Bool(true),
+		Description: aws.String("Parameter created/updated by SOPS publish command"),
```

Comment:
Since none of the other publishers currently populate a description, recommend removing this as part of this PR and if desired, add an enhancement request to populate a description across all of the publisher stores (where the backend supports it)